core/sugartypes.ml

```diff
@@ -1,42 +1,29 @@
 open CommonTypes
 open Operators
+open SourceCode
 open Utility
 
 (** The syntax tree created by the parser. *)
 
 type name = string [@@deriving show]
 
-type position = SourceCode.pos
-let dummy_position = SourceCode.dummy_pos
 
-let pp_position : Format.formatter -> position -> unit =
-  fun fmt _ -> Utility.format_omission fmt
 
-type 'a with_pos = { node : 'a
-                   ; pos  : position }
-                     [@@deriving show]
-let with_pos   pos node = { node; pos }
-let with_dummy_pos node = { node; pos = dummy_position }
-let tuple_of_with_pos {node; pos} = (node, pos)
+module Binder = struct
+  type t = (name * Types.datatype option) With_pos.t
+  [@@deriving show]
 
-(* A type alias to be used inside modules that define a node t and a with_pos
-   type, which is a node with attached position.  Alias is required due to
-   with_pos name inside the module overlapping with top-level with_pos type. *)
-module WithPos = struct
-  type 'a t = 'a with_pos [@@deriving show]
-end
+  let name b = let (n, _) = With_pos.node b in n
```

Comment:
Should be `to_name`.

Reply:
I personally feel like the `to_` convention should be reserved for type directed transformations, not necessarily simple projections like this. The `to_name` just somehow feels misleading here.

Reply:
You are transforming a term of type `t` into a term of type `name`. I think it is irrelevant whether the transformation is trivial.